frontend/src/components/collaborator/CollaboratorForm.vue

```diff
@@ -0,0 +1,139 @@
+              <span
+                >{{ item.role }} &middot;
+                <span class="grey--text"
+                  ><template v-for="icon in item.icons"
+                    ><v-icon :key="icon" x-small>{{ icon }}</v-icon
+                    >&thinsp;</template
+                  ></span
+                >
+              </span>
+            </template>
+          </e-select>
+        </div>
+      </template>
+    </v-tooltip>
+    <e-select
+      v-else
+      v-model="localCollaboration.role"
+      :name="$tc('entity.campCollaboration.fields.role')"
+      fieldname="role"
+      :items="items"
+      :hint="$tc('components.collaborator.collaboratorForm.roleHint')"
+      persistent-hint
+      item-value="key"
+      item-text="role"
+      vee-rules="required"
+    >
+      <template #item="{ item, on, attrs }">
+        <v-list-item v-bind="attrs" two-line v-on="on">
+          <v-list-item-content>
+            <v-list-item-title>{{ item.role }}</v-list-item-title>
+            <span class="caption">{{ item.abilities }}</span>
+          </v-list-item-content>
+          <v-list-item-action-text class="text-right">
+            <span
+              ><template v-for="icon in item.icons"
+                ><v-icon :key="icon" small>{{ icon }}</v-icon
+                >&thinsp;</template
+              ></span
+            >
+          </v-list-item-action-text>
+        </v-list-item>
+      </template>
+      <template #selection="{ item }">
+        <span
+          >{{ item.role }} &middot;
+          <span class="grey--text"
+            ><template v-for="icon in item.icons"
+              ><v-icon :key="icon" x-small>{{ icon }}</v-icon
+              >&thinsp;</template
+            ></span
+          >
+        </span>
+      </template>
+    </e-select>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'SettingsCollaboratorForm',
+  props: {
+    collaboration: { type: Object, required: true },
+    status: { type: [String, Boolean], required: false, default: false },
+    readonlyRole: { type: [String, Boolean], required: false, default: false },
+  },
+  computed: {
+    items() {
+      return [
+        {
+          key: 'manager',
+          role: this.$tc('entity.camp.collaborators.manager'),
+          abilities: this.$tc('global.collaborationAbilities.manager'),
+          icons: ['mdi-eye-outline', 'mdi-pencil-outline', 'mdi-cog-outline'],
+        },
+        {
+          key: 'member',
+          role: this.$tc('entity.camp.collaborators.member'),
+          abilities: this.$tc('global.collaborationAbilities.member'),
+          icons: ['mdi-eye-outline', 'mdi-pencil-outline'],
+        },
+        {
+          key: 'guest',
+          role: this.$tc('entity.camp.collaborators.guest'),
+          abilities: this.$tc('global.collaborationAbilities.guest'),
+          icons: ['mdi-eye-outline'],
+        },
+      ]
+    },
+    localCollaboration() {
+      return this.collaboration
+    },
```

Comment:
Couldn't you remove this? I think this just returns the reference to the prop object anyways, so you are not keeping a local copy here. So when you mutate the role via the form, you mutate `this.collaboration` directly.
So it makes no difference to keep a computed here, and the name evokes the wrong impression of a local copy.

Reply:
We use it this way already. This is a workaround for the eslint which would otherwise complain. This would need to be reworked fully. I'm not sure how to handle this sensible. #3500